bin/test

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env node
+
+'use strict';
+
+process.on('unhandledRejection', err => {
```

Comment:
I think this is wrong for the test suite.

The problem here is, that it prevents `chai` from testing rejections correctly. The tests and the run code must not set any `unhandledRejection` handler - more exactly, it must not turn rejections into exceptions.

It is very important that `chai-as-promised` continues after receiving an exception, because for `.to.be.rejectedWith()` tests the rejection will not be detected anymore by chai. This fails currently:
```
return expect(myFunctionThatShouldReject())
  .to.be.rejectedWith(/my error message/);
```

The rejection test is never reached, because someone forces the current test to be terminated.

Reply:
@HyperBrain this is about _unhandled_ promise rejections, so ones _not handled_, either by `chai-as-promised` or `mocha` or anything else.

What's in your example is promise that's obviously handled (passed to `expect` which via `rejectedWith` is handled by `chai-as-promised`), you can easily validate that by e.g. adding test as:

```javascript
  it.only('test', () =>
    expect(Promise.reject(new Error('Rejection'))).to.be.rejectedWith('Rejection'));
```
It'll pass as expected, and it won't be emitted as `unhandledRejection`

Example of _unhandled rejection_  would be a promise _not handled by any entity_, so: not passed to any promise processing function and not returned to any promise processing consumer, as e.g.:
```javascript
 it.only('test', () => {
  Promise.reject(new Error('Rejection'));
});
```